src/components/earn/PoolCardTri/ManageStake.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { TokenAmount, Token, ChainId } from '@trisolaris/sdk'
+import { useHistory } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+
+import { ButtonPrimary } from '../../Button'
+import Toggle from '../../Toggle'
+import { AutoRow } from '../../Row'
+import StakingModal from './StakingModalTri'
+import UnstakingModal from './UnstakingModalTri'
+
+import { useActiveWeb3React } from '../../../hooks'
+import { useWalletModalToggle } from '../../../state/application/hooks'
+
+import { BIG_INT_ZERO } from '../../../constants'
+
+import { StableSwapPoolName } from '../../../state/stableswap/constants'
+import { ChefVersions, EarnedNonTriRewards } from '../../../state/stake/stake-constants'
+import { MASTERCHEF_ADDRESS_V1, MASTERCHEF_ADDRESS_V2 } from '../../../state/stake/hooks-sushi'
+
+import { StyledMutedSubHeader } from './PoolCardTri.styles'
+
+type ManageStakeProps = {
+  stakedAmount: TokenAmount | null | undefined
+  isStaking: boolean
+  stableSwapPoolName?: StableSwapPoolName | null
+  tokens: Token[]
+  lpAddress: string
+  chefVersion: ChefVersions
+  poolId: number
+  lpToken: Token
+  noTriRewards: boolean
+  earnedNonTriRewards: EarnedNonTriRewards[]
+  earnedAmount?: TokenAmount
+  userLiquidityUnstaked?: TokenAmount
+  account?: string | null
+  isLegacy: boolean
+}
+
+function ManageStake({
+  stakedAmount,
+  isStaking,
+  stableSwapPoolName,
+  tokens,
+  lpAddress,
+  chefVersion,
+  poolId,
+  lpToken,
+  noTriRewards,
+  earnedNonTriRewards,
+  earnedAmount,
+  userLiquidityUnstaked,
+  account,
+  isLegacy
+}: ManageStakeProps) {
+  const toggleWalletModal = useWalletModalToggle()
+  const history = useHistory()
+  const { t } = useTranslation()
+
+  const userHasLiquidity = userLiquidityUnstaked?.greaterThan(BIG_INT_ZERO)
+
+  const [showStakingModal, setShowStakingModal] = useState(false)
+  const [showUnstakingModal, setShowUnstakingModal] = useState(false)
+  const [toggleIsStaking, setToggleIsStaking] = useState(isStaking || userHasLiquidity)
+
+  const stakingRewardAddress =
+    chefVersion === ChefVersions.V1 ? MASTERCHEF_ADDRESS_V1[ChainId.AURORA] : MASTERCHEF_ADDRESS_V2[ChainId.AURORA]
+
+  const enableDepositModal = useCallback(() => {
+    if (account) {
+      setShowStakingModal(true)
+    } else {
+      toggleWalletModal()
+    }
+  }, [account, toggleWalletModal])
+
+  function handleDepositClick(event: React.MouseEvent) {
+    enableDepositModal()
+    event.stopPropagation()
+  }
+
+  function handleWithDrawClick(event: React.MouseEvent) {
+    setShowUnstakingModal(true)
+    event.stopPropagation()
+  }
+
+  function handleToggle(event: React.MouseEvent) {
+    setToggleIsStaking(!toggleIsStaking)
+    event.stopPropagation()
```

Comment:
Shouldn't `event.stopPropagation()` come _first_ in all of these, in order to prevent race conditions?

Reply:
See above changes to `<Toggle />`